### ISSUE TITLE
gt_group breaks Quarto layout

### ISSUE THREAD

**Issue by allefeld** [rstudio/gt#1575]
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html) and [contributing guidelines](https://github.com/rstudio/gt/blob/master/.github/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/rstudio/gt/issues), whether open or closed, comment on the existing thread instead of posting a new issue.

## Description

Displaying a `gt_group` object from an R chunk in a Quarto document breaks the layout of the document

## Reproducible example

* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).
   
Quarto document:

``````
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

```{r}
library(gt)
h <- head(sp500, n = 3) |> gt()
t <- tail(sp500, n = 3) |> gt()
gt_group(h, t)
```

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
``````

Output:

![image](https://github.com/rstudio/gt/assets/1738353/567e2903-ae3c-4b42-8486-734151d9ad68)

Content after the display of `gt_group` disregard the page margins.

The reason is that in the produced HTML, after the second `table` element, the `main` element is closed:

```html
<td class="gt_row gt_right" headers="adj_close">16.66</td>
</tr>
</tbody>
</table>
</div>

</div>
</div>
</main></div>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>


<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
```

In the output HTML, each `table` is (together with a `style` element) wrapped in a `div`, and the two `div`s are wrapped in `<div class="cell-output-display">`. That means that after the `</table>` is closed and before the next `<p>`, there should be two `</div>`s. In the produced HTML, there is additionally `</div></main></div>`.

## Expected result

The additional `</div></main></div>` should not be there.

## Session info

End the reproducible example with a call to `sessionInfo()` in the same session (e.g. `reprex(session_info = TRUE)`) and include the output.

```
R version 4.3.2 (2023-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 12 (bookworm)

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-serial/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-serial/libopenblas-r0.3.21.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_US.UTF-8    LC_NUMERIC=C            LC_TIME=en_DK.UTF-8    
 [4] LC_COLLATE=en_US.UTF-8  LC_MONETARY=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8    LC_NAME=C               LC_ADDRESS=C           
[10] LC_TELEPHONE=C          LC_MEASUREMENT=C        LC_IDENTIFICATION=C    

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] gt_0.10.1

loaded via a namespace (and not attached):
 [1] digest_0.6.34     utf8_1.2.4        R6_2.5.1          fastmap_1.1.1    
 [5] tidyselect_1.2.0  xfun_0.41         magrittr_2.0.3    glue_1.7.0       
 [9] tibble_3.2.1      knitr_1.45        pkgconfig_2.0.3   htmltools_0.5.7  
[13] rmarkdown_2.25    dplyr_1.1.4       generics_0.1.3    lifecycle_1.0.4  
[17] xml2_1.3.6        cli_3.6.2         fansi_1.0.6       sass_0.4.8       
[21] vctrs_0.6.5       withr_3.0.0       compiler_4.3.2    tools_4.3.2      
[25] pillar_1.9.0      evaluate_0.23     rlang_1.1.3       jsonlite_1.8.8   
[29] htmlwidgets_1.6.4
```

**Comment by allefeld** [comment]
Of course this might also be a Quarto bug, but I haven't seen this behaviour in any other context, including display of a single `gt` object.

Quarto version is 1.4.549.

**Comment by allefeld** [comment]
I'm no longer certain this is related to `gt_group`, since the same happens with this attempted workaround:

``````
```{r}
#| results: hold
library(gt)
h <- head(sp500, n = 3) |> gt()
t <- tail(sp500, n = 3) |> gt()
h
t
```
``````

It does not happen without `results: hold`.

In case this is a Quarto bug: @cderv.

**Comment by cderv** [comment]
I think this was indeed a Quarto issue, but it has been fixed in 1.5 it seems.

If I try the latest pre release, it works ok. This is with 1.5.14


![image](https://github.com/rstudio/gt/assets/6791940/d4ad194b-cce0-4ec7-86c7-effb26b929ef)

**Comment by allefeld** [comment]
I can confirm it doesn't occur in 1.5.14. Thanks @cderv!
